File: src/types/workspacePrompt.ts
export interface WorkspacePrompt {
    id: string;
    workspace_id: string;
    title: string;
    blocks: string;
    created_at: string;
    updated_at: string;
}

export interface PromptBlock {
    type: 'text' | 'file';
    content?: string;
    fileId?: string;
    fileName?: string;
}

File: src-tauri/src/adapters/db/models/workspace_prompt.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct WorkspacePrompt {
    pub id: String,
    pub workspace_id: String,
    pub title: String,
    pub blocks: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
